Source/WebCore/dom/Element.cpp

```diff
@@ -3210,9 +3003,9 @@ void Element::removedFromAncestor(RemovalType removalType, ContainerNode& oldPar
 
     document().userActionElements().clearAllForElement(*this);
 
-    if (UNLIKELY(usesEffectiveTextDirection()) && !isValidDirValue(attributeWithoutSynchronization(dirAttr))) {
-        if (auto* parent = parentOrShadowHostElement(); !(parent && parent->usesEffectiveTextDirection()))
-            setUsesEffectiveTextDirection(false);
+    if (UNLIKELY(usesTextDirection()) && !hasValidTextDirectionality()) {
+        if (auto* parent = parentOrShadowHostElement(); !(parent && parent->usesTextDirection()))
+            setUsesTextDirection(false);
```

Comment:
Ditto about the old name being more clear.

Reply:
Done.